### PR TITLE
Add peerDependency warning if using < typescript 5

### PR DESCRIPTION
**Pull request by EliBates**
I discovered this issue using nextjs when when using typescript < 5. 
https://github.com/fabian-hiller/valibot/issues/352

This adds a peerDependency of typescript 5 so that anyone installing valibot will be warned if they don't have the appropriate version of TypeScript installed in their project.

### CODE REVIEW

**Comment by fabian-hiller** [comment]
The idea is great, but there is one problem. If someone is not using TypeScript, this change will erroneously display an error when installing Valibot.

**Comment by EliBates** [comment]
@fabian-hiller Good point. I think we can add:

```json
  "peerDependenciesMeta": {
    "typescript": {
      "optional": true
    }
  }
```

This should make it warn only users who are using typescript. I think this will be helpful because I still see a lot of projects on <5 typescript.

**Comment by fabian-hiller** [comment]
Thank you! I will review this PR in the next few days.

File: package.json [hunk 1]
```diff
@@ -10,6 +10,14 @@
     "url": "https://github.com/fabian-hiller/valibot"
   },
   "private": true,
+  "peerDependencies": {
+    "typescript": ">=5"
+  },
```
Comment (mxdvl) [inline package.json:15]: One thing to note about the TypeScript project is that [it does not follow SemVer](https://www.semver-ts.org/1-background.html#typescripts-versioning-policy), and breaking changes can be introduced in minor versions.

You might want to be more explicit about versions that are compatible with the tilde syntax:

```suggestion
  "peerDependencies": {
    "typescript": "~5.0||~5.1||~5.2||~5.3"
  },
```
Or even just a more specific range:

```suggestion
  "peerDependencies": {
    "typescript": "=5"
  },
```

**Comment by fabian-hiller** [comment]
Sorry for not picking up this PR yet. Other things had a higher priority. I expect to review it this month.

**Comment by fabian-hiller** [comment]
Unfortunately Git screwed something up. I will close this PR and add the changes to the source myself. I will add you as a co-author to the commit.

**Pull request closed without merging**